qiskit/chemistry/transformations/fermionic_transformation.py

```diff
@@ -0,0 +1,622 @@
+                if len(self._z2symmetry_reduction) != len(z2_symmetries.symmetries):
+                    raise QiskitChemistryError('z2symmetry_reduction tapering values list has '
+                                               'invalid length {} should be {}'.
+                                               format(len(self._z2symmetry_reduction),
+                                                      len(z2_symmetries.symmetries)))
+                valid = np.all(np.isin(self._z2symmetry_reduction, [-1, 1]))
+                if not valid:
+                    raise QiskitChemistryError('z2symmetry_reduction tapering values list must '
+                                               'contain -1\'s and/or 1\'s only was {}'.
+                                               format(self._z2symmetry_reduction,))
+                z2_symmetries.tapering_values = self._z2symmetry_reduction
+
+            logger.debug('Apply symmetry with tapering values %s', z2_symmetries.tapering_values)
+            chop_to = 0.00000001  # Use same threshold as qubit mapping to chop tapered operator
+            z2_qubit_op = z2_symmetries.taper(qubit_op).chop(chop_to)
+            z2_aux_ops = []
+            for aux_op in aux_ops:
+                if aux_op is None:
+                    z2_aux_ops += [None]
+                else:
+                    z2_aux_ops += [z2_symmetries.taper(aux_op).chop(chop_to)]
+
+        return z2_qubit_op, z2_aux_ops, z2_symmetries
+
+    @staticmethod
+    def _check_commutes(cliffords: List[WeightedPauliOperator],
+                        operator: WeightedPauliOperator) -> bool:
+        """
+        Check commutations
+
+        Args:
+            cliffords : cliffords
+            operator: qubit operator
+
+        Returns:
+            Boolean: does_commute
+        """
+        commutes = []
+        for clifford in cliffords:
+            commutes.append(operator.commute_with(clifford))
+        does_commute = np.all(commutes)
+        logger.debug('  \'%s\' commutes: %s, %s', operator.name, does_commute, commutes)
+        return does_commute
+
+    def get_default_filter_criterion(self) -> Optional[Callable[[Union[List, np.ndarray], float,
+                                                                 Optional[List[float]]], bool]]:
+        """Returns a default filter criterion method to filter the eigenvalues computed by the
+        eigen solver. For more information see also
+        aqua.algorithms.eigen_solvers.NumPyEigensolver.filter_criterion.
+
+        In the fermionic case the default filter ensures that the number of particles is being
+        preserved.
+        """
+
+        # pylint: disable=unused-argument
+        def filter_criterion(self, eigenstate, eigenvalue, aux_values):
+            # the first aux_value is the evaluated number of particles
+            num_particles_aux = aux_values[0][0]
+            return np.isclose(sum(self.molecule_info['num_particles']), num_particles_aux)
```

Comment:
We ought to be able to match on the spin as well I think - particles is only one aspect. Particles plus spin should be the right filter I think.

Reply:
I agree with you. The reason we have the filter now is for the bosonic case and we need to measure only the number of particles (for this particular case). We can keep this as backlog I think for the upcoming development

Reply:
It should be simple enough to fix this. I thought this was the for Numpy based Excited state solver on the Fermionic side where this filter arguably is needed too.